specification/azsadmin/resource-manager/containerregistry/Microsoft.ContainerRegistry.Admin/preview/2019-11-01/capacity.json

```diff
@@ -0,0 +1,180 @@
+          },
+          {
+            "$ref": "containerregistry.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CapacityNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Container registry capacity property is being returned.",
+            "schema": {
+              "$ref": "#/definitions/ContainerRegistryCapacity"
+            }
+          },
+          "default": {
+            "description": "Error Response.",
+            "schema": {
+              "$ref": "containerregistry.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerRegistry.Admin/locations/{location}/capacity": {
+      "get": {
+        "x-ms-examples": {
+          "Returns container registry capacity properties.": {
+            "$ref": "./examples/Configuration/ListCapacity.json"
+          }
+        },
+        "tags": [
+          "ContainerRegistryCapacities"
+        ],
+        "description": "Returns a list of container registry capacity properties.",
+        "operationId": "ContainerRegistryCapacity_List",
+        "parameters": [
+          {
+            "$ref": "containerregistry.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "containerregistry.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "containerregistry.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Container registry capacity properties are being returned.",
+            "schema": {
+              "$ref": "#/definitions/ContainerRegistryCapacityList"
+            }
+          },
+          "default": {
+            "description": "Error Response.",
+            "schema": {
+              "$ref": "containerregistry.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ContainerRegistryCapacity": {
+      "description": "Container registry capacity property.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Container registry capacity property.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ContainerRegistryCapacityProperty"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "containerregistry.json#/definitions/Resource"
+        }
+      ]
+    },
+    "ContainerRegistryCapacityProperty": {
+      "description": "Container registry capacity property.",
+      "type": "object",
+      "properties": {
+        "maximumCapacityInGb": {
+          "description": "Total storage capacity (GB) which can used by the registry.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "allowPush": {
```

Comment:
booleans are an anti-pattern, because they often don't describe both states.  It would be better to have a value like 'pushStatus' with enum values 'allowed' and 'blocked'

Reply:
It is ok for this case. we follow the Azure Container registry code base.